osbuild/sources.py

```diff
@@ -4,13 +4,73 @@
 import os
 import json
 import tempfile
-from typing import List
+from typing import List, Dict
+from enum import Enum
 
+from osbuild.formats_common import OSBuildError
 from . import host
 from .objectstore import ObjectStore
 from .util.types import PathLike
 
 
+class SourceErrorKind(Enum):
+    UNKNOWN = 0
+    CHECKSUM = 1
+    # OSTree errors
+    OSTREE_INIT = 2
+    OSTREE_CONFIG = 3
+    OSTREE_REMOTE_ADD = 4
+    OSTREE_GPG_IMPORT = 5
+    OSTREE_PULL = 6
+    OSTREE_REMOTE_DELETE = 7
+    # Skopeo errors
+    SKOPEO_COPY = 8
+    SKOPEO_INSPECT = 9
+    # Curl errors
+    CURL_BAD_INPUT = 10
+    CURL_INTERNAL_ERROR = 11
+    CURL_NETWORKING = 12
+    CURL_CREDENTIALS = 13
+    CURL_STORAGE = 14
+    CURL_SECURITY = 15
```

Comment:
I think using strings (on reverse domain name notation) rather than ints would make this more self-describing and also more in line with how we usually do enums in osbuild.

Another thought: if we made these nested we could perhaps define the errors specific to each source in the source itself? Essentially how we treat regular results, `sources.py` is not aware of what sources we have actually defined.